### ISSUE TITLE
What is the status of this node-qrpc?

### ISSUE THREAD

**Issue by jacoscaz** [andrasq/node-qrpc#2]
Hello! I'm evaluating different RPC libraries for a new projects and came across `qrpc` via NPM, which looks really nice (kudos to you @andrasq). The last published version on NPM dates back to 4 years ago but commit activity here on GitHub suggests that the library is still being somewhat maintained. Test targets in the `.travis.yml` file cover up to Node.js 15.x, also another indicator of life. What is the general status of `node-qrpc`? Is it being actively maintained?

**Comment by andrasq** [comment]
Hi, I haven't abandoned qrpc, been working on other projects (database drivers, data serialization, currently pubsub).  Qrpc is one of my favorites, so if you have questions or need features I would gladly help.

**Comment by jacoscaz** [comment]
Hello!

Good to know that qrpc is still maintained, thank you for getting back to me so quickly. I actually enjoy stable modules without too much activity on them, I think stability is a highly underrated features.

May I ask what your perspective is on Typescript and whether you have plans to add type-safe support to qrpc?

Best regards,
Jacopo.

---

Jacopo Scazzosi

**Comment by andrasq** [comment]
Hi Jacopo, you mean support as in adding a `.d.ts` file with types and call signatures?  That sounds fine, it shouldn't be too hard to cover the major subsystems.
Andras

**Comment by MarcWeber** [comment]
You can also port to .ts and have typescript compile down to .js and .d.ts type files. What about browser support ?

**Comment by andrasq** [comment]
True, could rewrite it, but would be a lateral move at best.  Adding typing offers more bang for the buck.
As to browser support, I figured I could contribute to fast non-blocking server-to-server (or server-to-multi-tenant-client) communication.  I don't know enough about the browser side to understand how a browser would run services, or how to multiplex calls onto a raw tcp/ip socket, how to make it portable, or even whether that would be faster than, say, simply batching calls.
